packages/ts/src/rules/withStatements.ts

```diff
@@ -0,0 +1,38 @@
+import { typescriptLanguage } from "../language.js";
+
+const withKeyword = "with";
+
+export default typescriptLanguage.createRule({
+	about: {
+		description: "Reports using with statements",
+		id: "withStatements",
+		preset: "stylistic",
```

Comment:
I originally filed this as `stylistic`, but the more I think about it, I'm starting to lean toward `logical`. It's _also_ a performance issue ([MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/with#:~:text=makes%20optimization%20impossible)), but I think it's better to keep it in the main TS plugin so it's available for stylistic/logical reasons.

Reply:
Oh, right, TypeScript bans `with` statements! That means this would actually be _untyped_. I'll make that switch, sorry for the back-and-forth.